pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -322,14 +287,18 @@ func (c *controller) Get(typ, name, namespace string) *model.Config {
 	}
 
 	config, err := crd.ConvertObject(s, obj, c.client.domainSuffix)
-	if err == nil && features.EnableCRDValidation.Get() {
-		err = s.Validate(config.Name, config.Namespace, config.Spec)
-	}
-
 	if err != nil {
 		return nil
 	}
 
+	if features.EnableCRDValidation.Get() {
+		if err = s.Validate(config.Name, config.Namespace, config.Spec); err != nil {
+			// This is to prevent too verbose
+			log.Debugf("%s [%s/%s] is invalid: %v", typ, config.Namespace, config.Name, err)
```

Comment:
Can we increment k8serrors metric here as well similar to how we do in `handleValidationFailure`? Also can we make the log level and message be in sync with what is being done in `handleValidationFailure` -  I do not have a strong opinion on which one to be used (debug/warn) but better to be consistent?

Reply:
Agreed, this log level should be consistent